sklearn/metrics/tests/test_common.py

```diff
@@ -1550,13 +1550,15 @@ def check_sample_weight_invariance(name, metric, y1, y2):
     ),
 )
 def test_regression_sample_weight_invariance(name):
-    n_samples = 50
+    n_samples = 51
     random_state = check_random_state(0)
     # regression
     y_true = random_state.random_sample(size=(n_samples,))
     y_pred = random_state.random_sample(size=(n_samples,))
+    sample_weight = np.arange(len(y_true))
```

Comment:
Changed as suggested in https://github.com/scikit-learn/scikit-learn/pull/30787#discussion_r1952568151. This works great.

I've amended `check_sample_weight_invariance` to optionally take `sample_weight` as I didn't want to change the `sample_weight` for all tests that use `check_sample_weight_invariance`, WDYT @glemaitre ?